### ISSUE TITLE
Redirecting from http to https in nanohttpd

### ISSUE THREAD

**Issue by JiashuGuo** [NanoHttpd/nanohttpd#538]
Hi there, I am trying to automatically correct/add https at the beginning of the URL the user typed in, for example: no matter user type http://XXXX or just XXXX I would like to change it to https://XXXX. 

Does any one know if nanohttpd support redirecting from http request to https? I have viewed lots of answer online and most of them related to use ./htaccess in Apache server, I am afraid Nanohttpd does't support that method.

Thanks for any discussion!

**Comment by LordFokas** [comment]
You need to be running two servers for that, because HTTP and HTTPS use different ServerSocket implementations, so they need to be on different ports (80 & 443 are the defaults).

This being done, you can simply have the HTTP server redirect everyone (using HTTP 3XX with `Location` headers) to the same path in the HTTPS server and you're set.

**Comment by JiashuGuo** [comment]
Thank you so much for the idea, I am trying with this in the App.

**Comment by JiashuGuo** [comment]
> You need to be running two servers for that, because HTTP and HTTPS use different ServerSocket implementations, so they need to be on different ports (80 & 443 are the defaults).
> 
> This being done, you can simply have the HTTP server redirect everyone (using HTTP 3XX with `Location` headers) to the same path in the HTTPS server and you're set.

Thank you so much for the idea, I am trying with this in the App.

**Comment by JiashuGuo** [comment]
> You need to be running two servers for that, because HTTP and HTTPS use different ServerSocket implementations, so they need to be on different ports (80 & 443 are the defaults).
> 
> This being done, you can simply have the HTTP server redirect everyone (using HTTP 3XX with `Location` headers) to the same path in the HTTPS server and you're set.

I tried to run with two server, the redirecting works well, but can't on default port, I have add redirect in IP table still seems doesn't work. Do you have any idea about running server under port 1024 in Linux?

**Comment by NoahAndrews** [comment]
Are you running the application as root? By default, processes need root privileges to bind to ports 1-1024 on Linux.

**Comment by JiashuGuo** [comment]
No, we don’t run it on as root for security consider, so I am looking for
other solution like redirect in ip table.

**Comment by JiashuGuo** [comment]
The iptable redirecting works after I reboot the device.

**Comment by LordFokas** [comment]
Either use ports above 1024 or `sudo`. You don't really have other choices.
